functions/common/datastore.js

```diff
@@ -66,12 +66,37 @@ class Datastore {
 
   fetchForm(projectId, featureTypeId, formId) {
     return this.fetchDoc_(
-        `projects/${projectId}/featureTypes/${featureTypeId}/forms/${formId}`);
+      `projects/${projectId}/featureTypes/${featureTypeId}/forms/${formId}`);
   }
 
   fetchSheetsConfig(projectId) {
     return this.fetchDoc_(`projects/${projectId}/sheets/config`);
   }
+
+  insertFeature(projectId, layerId, { caption, location }) {
+    var docRef = this.db_.collection('projects').doc(projectId);
+    return docRef.get().then((docSnapshot) => {
+      if (docSnapshot.exists) {
+        docRef.collection('features').add({
```

Comment:
We'd want to return the `Promise` that gets returned here so upstream callers can wait for the insert to complete.